addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClientV2.java

```diff
@@ -0,0 +1,258 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.evohome.internal.api;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.evohome.internal.api.models.v2.request.HeatSetPoint;
+import org.openhab.binding.evohome.internal.api.models.v2.request.HeatSetPointBuilder;
+import org.openhab.binding.evohome.internal.api.models.v2.request.Mode;
+import org.openhab.binding.evohome.internal.api.models.v2.request.ModeBuilder;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Authentication;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.UserAccount;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the evohome client V2 api
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeApiClientV2 implements EvohomeApiClient {
+
+    private static final String APPLICATION_ID = "b013aa26-9724-4dbd-8897-048b9aada249";
+
+    private final Logger logger = LoggerFactory.getLogger(EvohomeApiClientV2.class);
+
+    private final SslContextFactory sslContextFactory = new SslContextFactory();
+    private final HttpClient httpClient = new HttpClient(sslContextFactory);
```

Comment:
You should not instantiate your own `HttpClient`, but rather use the shared client that can be obtained through the `HttpClientFactory` (see [here](https://www.eclipse.org/smarthome/documentation/development/bindings/dependencies.html#for-http-operations)). You can depend on that service in your handler factory and then reach the http client instance through the constructors, so that it ends up here.